apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransactionOptions.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License"):
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+
+namespace Google.Cloud.Spanner.Data;
+
+/// <summary>
+/// Options that may be applied to a transaction after creation, usually before commiting the transaction
+/// or before executing the first transactional statement.
```

Comment:
Possibly explain that this is mutable (compared with SpannerTransactionCreationOptions) and under what circumstances mutations are observed? Or is the plan to put this in the conceptual docs?

Reply:
I'll add here, even if we add some of this elsewhere, it doesn't harm to have it in here.